### PR TITLE
JTJ-58-FIle-Upload

### PR DESCRIPTION
**Pull request by tgi-wtag**
A really short PR i added the Functionality to upload files to a journal entry using active storage.

https://github.com/tgi-wtag/Journey-tg/assets/141820846/2ea2cb12-1ff3-45b6-bec5-30da2cd033c2

Thank you for reviewing ! 🌻

### CODE REVIEW

File: spec/models/journal_entry_spec.rb [hunk 1]
```diff
@@ -18,4 +19,31 @@
   describe 'associations' do
     it { should belong_to(:user) }
   end
+  let(:user) { create(:user) }
+
+  describe 'JournalEntry' do
+    it 'creates an attachment' do
+      journal_entry_with_document = FactoryBot.create(:journal_entry, user: user)
+      file_path = Rails.root.join('spec/support/pdf_for_testing.pdf')
+      journal_entry_with_document.attached_document.attach(io: File.open(file_path), filename: 'pdf_for_testing.pdf', content_type: 'application/pdf')
```
Comment (chm-wtag) [inline spec/models/journal_entry_spec.rb:28]: study about `trait` in rspec factory bot and see if you can add attached documents while creating factory instances.

**Review by chm-wtag — APPROVED** [review]

File: app/models/journal_entry.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class JournalEntry < ApplicationRecord
   belongs_to :user
+  has_one_attached :attached_document
```
Comment (evk-wtag) [inline app/models/journal_entry.rb:3]: Could you rename it to just `document`?

File: spec/models/journal_entry_spec.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 require 'rails_helper'
 
 RSpec.describe JournalEntry, type: :model do
+
```
Comment (evk-wtag) [inline spec/models/journal_entry_spec.rb:4]: Let's remove this line

File: app/models/journal_entry.rb (hunk 2 above)
Comment (evk-wtag) [reply app/models/journal_entry.rb:3]: Also let's add a validation for it's acceptable content types

File: spec/models/journal_entry_spec.rb (hunk 1 above)
Comment (tgi-wtag) [reply spec/models/journal_entry_spec.rb:28]: ok thank you i will create a backend story regarding it and study it in the future, but i wont address it in this PR given my Deadline for the application is in 2 days.

File: app/models/journal_entry.rb (hunk 2 above)
Comment (tgi-wtag) [reply app/models/journal_entry.rb:3]: it is about ataching documents to a journal entry so i believe this name describes it pretty well, regarding validating content types i will ad a story in my backlog, but Due to the deadline in 2 days i wont add this in this PR, but thank you for the recomendation.

**Review by lra-wtag — APPROVED** [review]

**Pull request merged**